marshal.go

```diff
@@ -1579,7 +2113,8 @@ func writeCollectionSize(info CollectionType, n int, buf *bytes.Buffer) error {
 	return nil
 }
 
-func marshalList(info TypeInfo, value interface{}) ([]byte, error) {
+// Marshal marshals the value for the given TypeInfo into a byte slice.
+func (listSetCQLType) Marshal(info TypeInfo, value interface{}) ([]byte, error) {
 	listInfo, ok := info.(CollectionType)
```

Comment:
The fact that this implementation relies on accessing `listInfo.proto` which isn't exported shows that there needs to be a better (and accessible) way of knowing what protocol version is being used otherwise user defined registrations won't be able to do this.